### DISCUSSION TITLE
Themes: Allow alternative suffixes for template files

### DISCUSSION THREAD

**Discussion by traits** [getpelican/pelican#3088] [Ideas]
Could the allowed file suffixes for theme templates be extended to something more jinja-specific - for example .jinja2? 

This would allow somewhat better support in editors, IDE's etc.. For example: Visual Studio Code has all the plugins for jinja file support, but difficulties to separate this from generic html handling because the suffix is the same. You can get syntax highlighting, intendation and intellisense. But not all of them together as far as I can see. This could be simply solved by a more relaxed suffix policy.

**Comment by galaxy4public** [comment]
What prevents you from creating symbolic links with the extensions you desire? E.g. in your theme's templates directory you can, say, create a symbolic link `article.html.jinja2` -> `article.html` and edit `article.html.jinja2` (the majority of decent editors will respect a symbolic link and will edit the correct file).  Symbolic links are supported on all mainstream operating systems, so should not be an issue.

**Reply by traits** [reply]
> Symbolic links are supported on all mainstream operating systems, so should not be an issue.

On Windows, this is handled by a policy that by default requires administrator roles to create links, for example. This can be changed, but links on this operating system have always been special. I would like to avoid this.

**Reply by lahwaacz** [reply]
The symlinks approach is against the DRY principle.

That said, VSCode allows to configure per-project overrides for file type autodetection. You can put this in `project/.vscode/settings.json`:
```json
{
    "files.associations": {
        "**/templates/*.html": "jinja-html"
    }
}
```
